spec/services/tax_tribs/mapping_code_determiner_spec.rb

```diff
@@ -31,12 +31,14 @@ def closure_case_type; end
 
   context 'when there is no case_type nor closure_case_type' do
     it { is_expected.to fail_to_determine_mapping_code }
```

Comment:
Looking at it, and seeing what your specs add, I think the spec above can be simplified from:

```
tax-tribunals-datacapture/spec/support/mapping_code_determiner_matcher.rb:14

RSpec::Matchers.define :fail_to_determine_mapping_code do
  match do |mapping_code_determiner|
    return false if mapping_code_determiner.valid_for_determining_mapping_code?

    begin
      mapping_code_determiner.mapping_code
      false
    rescue
      true
    end
  end

  failure_message do |mapping_code_determiner|
    "expected mapping code determiner to raise an error when determining mapping code, but got '#{mapping_code_determiner.mapping_code}'."
  end
end

AND 

it { is_expected.to fail_to_determine_mapping_code }
```
To:

```
it { expect{ subject.mapping_code }.to raise_error(<WhateverError>) }
```